micronota/parsers/embl.py

```diff
@@ -0,0 +1,432 @@
+    return _construct(record, RNA, **kwargs)
+
+
+def _construct(record, constructor=None, **kwargs):
+    seq, md, pmd = record
+    if constructor is None:
+        unit = md['ID']['unit'].lower()
+        if unit == 'bp':
+            constructor = DNA
+        elif unit == 'aa':
+            constructor = Protein
+    if constructor == RNA:
+        return DNA(
+            seq, metadata=md, positional_metadata=pmd, **kwargs).transcribe()
+    else:
+        return constructor(
+            seq, metadata=md, positional_metadata=pmd, **kwargs)
+
+
+def _parse_records(fh, parser):
+    data_chunks = []
+    for line in _line_generator(fh, skip_blanks=True, strip=False):
+        if line.startswith('//'):
+            yield parser(data_chunks)
+            data_chunks = []
+        else:
+            data_chunks.append(line)
+
+
+def _parse_single_embl(chunks):
+    metadata = {}
+    positional_metadata = None
+    sequence = ''
+
+    def split(s):
+        if s[0].isspace():
+            return None, s.strip()
+        else:
+            return s.split(None, 1)
+
+    for header, data in _yield_section(chunks, split, strip=False):
+        if header in ['XX', 'RN', 'RC', 'RP', 'RX', 'RG', 'RA', 'RT', 'RL']:
+            continue
+        parser = _PARSER_TABLE.get(header)
+        if parser:
+            data = parser(data)
+        if header == 'SQ':
+            sequence = data
+        else:
+            metadata[header] = data
+
+    return sequence, metadata, positional_metadata
+
+
+def _parse_id(lines):
+    '''Parse ID line.'''
+    items = lines[0].rstrip('.').split(';')
+    res = dict()
+    # quality would be "reviewed" or "unreviewed"
+    res['id'], res['quality'] = items[0].split()
+    res['size'], res['unit'] = items[-1].split()
+    return res
+
+
+def _parse_ac(lines):
+    return lines[0].rstrip(';')
+
+
+def _parse_oc(lines):
+    return ' '.join(lines).rstrip('.')
+
+
+def _parse_ox(lines):
+    line = lines[0]
+    taxon_id = line.replace('NCBI_TaxID=', '').rstrip(';')
+    return taxon_id
+
+
+def _parse_oh(lines):
+    '''Parse the OH (Organism Host) line.
+
+    Optional and appears only in viral entries. It indicates the host
+    organism(s) that are susceptible to be infected by a virus.
+
+    Example line:
+    'NCBI_TaxID=8295; Ambystoma (mole salamanders).'
+    '''
+    taxa = dict()
+    for line in lines:
+        taxon_id, taxon = line.split(';')
+        taxon_id = taxon_id.replace('NCBI_TaxID=', '')
+        taxon = taxon.strip(' .')
+        taxa[taxon_id] = taxon
+    return taxa
+
+
+def _parse_dr(lines):
+    '''Parse the DR (Database Cross-reference) line.'''
+    res = dict()
+    for db_name, db_lines in _yield_section(lines, lambda s: s.split('; ', 1)):
+        res[db_name] = db_lines
+    return res
+
+
+def _parse_pe(lines):
+    return lines[0].split(':')[0]
+
+
+def _parse_kw(lines):
+    l = ' '.join(lines).rstrip('.')
+    return l.split('; ')
+
+
+def _parse_sq(lines):
```

Comment:
will this handle all EMBL files? For instance, take a look at the `SQ` line here as it contains additional information about the sequence like the number of nucleotides in each line

http://www.ebi.ac.uk/ena/data/view/Non-coding:JN580308.1:8296..8365:tRNA&display=text